pkg/cluster/hive.go

```diff
@@ -34,14 +34,24 @@ func (m *manager) hiveEnsureResources(ctx context.Context) error {
 	return m.hiveClusterManager.CreateOrUpdate(ctx, m.subscriptionDoc, m.doc)
 }
 
-func (m *manager) hiveClusterDeploymentReady(ctx context.Context) (bool, error) {
+func (m *manager) hiveClusterDeploymentReady(ctx context.Context) (bool, bool, error) {
 	m.log.Info("waiting for cluster deployment to become ready")
-	return m.hiveClusterManager.IsClusterDeploymentReady(ctx, m.doc)
+	ok, err := m.hiveClusterManager.IsClusterDeploymentReady(ctx, m.doc)
+	retry := true
+	if err != nil {
+		retry = false
+	}
+	return ok, retry, err
```

Comment:
I think we can simplify the function if we make use of named return parameters + we remove the temporal _retry_ variable (we also return as soon as possible in case of an error). Something like:
```go
func (m *manager) hiveClusterDeploymentReady(ctx context.Context) (deploymentReady bool, retry bool, err error) {
	m.log.Info("waiting for cluster deployment to become ready")
	deploymentReady, err = m.hiveClusterManager.IsClusterDeploymentReady(ctx, m.doc)
	if err != nil {
		return deploymentReady, false, err
	}
	return deploymentReady, true, err
}
```